### ISSUE TITLE
Humanpokesnipe not looting pokestops

### ISSUE THREAD

**Issue by Capusjon** [NoxxDev/NecroBot#894]
When the bot is walking towards snipetargets it's not looting pokestops. Also I think it doesnt check for lured pokemon on the sniping locations either. 

Mabey because it doesn't always loot the pokestop?

### Expected Behavior
looting pokestops and finding snipetargets

### Actual Behavior
not looting pokestops and not finding snipetargets (possibly because they are lured?)

### Your config.json
```
 },
  "LocationConfig": {
    "DisableHumanWalking": false,
    "StartFromLastPosition": true,
    "DefaultLatitude": 52.068351,
    "DefaultLongitude": 4.218489,
    "WalkingSpeedInKilometerPerHour": 6.16,
    "UseWalkingSpeedVariant": true,
    "WalkingSpeedVariant": 1.2,
    "ShowVariantWalking": false,
    "RandomlyPauseAtStops": true,
    "MaxSpawnLocationOffset": 10,
    "MaxTravelDistanceInMeters": 650
  },
  "TelegramConfig": {
    "UseTelegramAPI": false,
    "TelegramAPIKey": null,
    "TelegramPassword": null
  },
  "GPXConfig": {
    "UseGpxPathing": true,
    "GpxFile": "kijkduin5.GPX"
  },
  "SnipeConfig": {
    "UseSnipeLocationServer": false,
    "SnipeLocationServer": "localhost",
    "SnipeLocationServerPort": 16969,
    "GetSniperInfoFromPokezz": false,
    "GetOnlyVerifiedSniperInfoFromPokezz": true,
    "GetSniperInfoFromPokeSnipers": true,
    "GetSniperInfoFromPokeWatchers": true,
    "GetSniperInfoFromSkiplagged": true,
    "MinPokeballsToSnipe": 20,
    "MinPokeballsWhileSnipe": 0,
    "MinDelayBetweenSnipes": 60000,
    "SnipingScanOffset": 0.005,
    "SnipeAtPokestops": false,
    "SnipeIgnoreUnknownIv": false,
    "UseTransferIvForSnipe": false,
    "SnipePokemonNotInPokedex": false,
    "UseSnipeLimit": true,
    "SnipeRestSeconds": 600,
    "SnipeCountLimit": 39
  },
  "HumanWalkSnipeConfig": {
    "Enable": true,
    "DisplayPokemonList": true,
    "MaxDistance": 1500.0,
    "MaxEstimateTime": 900.0,
    "CatchEmAllMinBalls": 10,
    "TryCatchEmAll": true,
    "CatchPokemonWhileWalking": true,
    "SpinWhileWalking": true,
    "AlwaysWalkback": false,
    "SnipingScanOffset": 0.025,
    "WalkbackDistanceLimit": 300.0,
    "IncludeDefaultLocation": true,
    "UsePokeRadar": true,
    "UseSkiplagged": true,
    "UseSnipePokemonList": true,
    "MaxSpeedUpSpeed": 6.0,
    "AllowSpeedUp": true,
    "DelayTimeAtDestination": 10000,
    "UseFastPokemap": true,
    "UsePogoLocationFeeder": true
  },
  "PokeStopConfig": {
    "UsePokeStopLimit": true,
    "PokeStopLimit": 1998,
    "PokeStopLimitMinutes": 1470
  },
irrelevant stuff
 ],
  "HumanWalkSnipeFilters": {
    "Dragonite": {
      "MaxDistance": 3000.0,
      "Priority": 1,
      "MaxWalkTimes": 900.0,
      "CatchPokemonWhileWalking": false,
      "SpinPokestopWhileWalking": true,
      "MaxSpeedUpSpeed": 6.0,
      "AllowSpeedUp": true,
      "DelayTimeAtDestination": 0
    },
    "Dragonair": {
      "MaxDistance": 3000.0,
      "Priority": 1,
      "MaxWalkTimes": 900.0,
      "CatchPokemonWhileWalking": false,
      "SpinPokestopWhileWalking": true,
      "MaxSpeedUpSpeed": 6.0,
      "AllowSpeedUp": true,
      "DelayTimeAtDestination": 0
    },
    "Dratini": {
      "MaxDistance": 2000.0,
      "Priority": 1,
      "MaxWalkTimes": 900.0,
      "CatchPokemonWhileWalking": false,
      "SpinPokestopWhileWalking": true,
      "MaxSpeedUpSpeed": 6.0,
      "AllowSpeedUp": true,
      "DelayTimeAtDestination": 0
    },
    "Charizard": {
      "MaxDistance": 3000.0,
      "Priority": 1,
      "MaxWalkTimes": 900.0,
      "CatchPokemonWhileWalking": false,
      "SpinPokestopWhileWalking": true,
      "MaxSpeedUpSpeed": 6.0,
      "AllowSpeedUp": true,
      "DelayTimeAtDestination": 0
    },
    "Snorlax": {
      "MaxDistance": 3000.0,
      "Priority": 1,
      "MaxWalkTimes": 900.0,
      "CatchPokemonWhileWalking": false,
      "SpinPokestopWhileWalking": true,
      "MaxSpeedUpSpeed": 6.0,
      "AllowSpeedUp": true,
      "DelayTimeAtDestination": 0
    },
    "Lapras": {
      "MaxDistance": 3000.0,
      "Priority": 1,
      "MaxWalkTimes": 900.0,
      "CatchPokemonWhileWalking": false,
      "SpinPokestopWhileWalking": true,
      "MaxSpeedUpSpeed": 6.0,
      "AllowSpeedUp": true,
      "DelayTimeAtDestination": 0
    },
    "Exeggutor": {
      "MaxDistance": 1500.0,
      "Priority": 3,
      "MaxWalkTimes": 600.0,
      "CatchPokemonWhileWalking": false,
      "SpinPokestopWhileWalking": true,
      "MaxSpeedUpSpeed": 6.0,
      "AllowSpeedUp": false,
      "DelayTimeAtDestination": 0
    },
    "Vaporeon": {
      "MaxDistance": 1100.0,
      "Priority": 3,
      "MaxWalkTimes": 800.0,
      "CatchPokemonWhileWalking": false,
      "SpinPokestopWhileWalking": true,
      "MaxSpeedUpSpeed": 6.0,
      "AllowSpeedUp": false,
      "DelayTimeAtDestination": 0
    },
    "Lickitung": {
      "MaxDistance": 1800.0,
      "Priority": 2,
      "MaxWalkTimes": 800.0,
      "CatchPokemonWhileWalking": false,
      "SpinPokestopWhileWalking": true,
      "MaxSpeedUpSpeed": 6.0,
      "AllowSpeedUp": false,
      "DelayTimeAtDestination": 0
    },
    "Chansey": {
      "MaxDistance": 1500.0,
      "Priority": 2,
      "MaxWalkTimes": 800.0,
      "CatchPokemonWhileWalking": false,
      "SpinPokestopWhileWalking": true,
      "MaxSpeedUpSpeed": 6.0,
      "AllowSpeedUp": false,
      "DelayTimeAtDestination": 0
    },
    "Hitmonlee": {
      "MaxDistance": 1500.0,
      "Priority": 3,
      "MaxWalkTimes": 800.0,
      "CatchPokemonWhileWalking": false,
      "SpinPokestopWhileWalking": true,
      "MaxSpeedUpSpeed": 6.0,
      "AllowSpeedUp": false,
      "DelayTimeAtDestination": 0
    },
    "Kangaskhan": {
      "MaxDistance": 800.0,
      "Priority": 4,
      "MaxWalkTimes": 800.0,
      "CatchPokemonWhileWalking": true,
      "SpinPokestopWhileWalking": true,
      "MaxSpeedUpSpeed": 6.0,
      "AllowSpeedUp": false,
      "DelayTimeAtDestination": 0
    },
    "Gyarados": {
      "MaxDistance": 1800.0,
      "Priority": 2,
      "MaxWalkTimes": 800.0,
      "CatchPokemonWhileWalking": false,
      "SpinPokestopWhileWalking": true,
      "MaxSpeedUpSpeed": 6.0,
      "AllowSpeedUp": true,
      "DelayTimeAtDestination": 0
    },
    "Blastoise": {
      "MaxDistance": 3000.0,
      "Priority": 1,
      "MaxWalkTimes": 900.0,
      "CatchPokemonWhileWalking": false,
      "SpinPokestopWhileWalking": true,
      "MaxSpeedUpSpeed": 6.0,
      "AllowSpeedUp": true,
      "DelayTimeAtDestination": 0
    },
    "Venusaur": {
      "MaxDistance": 3000.0,
      "Priority": 1,
      "MaxWalkTimes": 900.0,
      "CatchPokemonWhileWalking": false,
      "SpinPokestopWhileWalking": true,
      "MaxSpeedUpSpeed": 6.0,
      "AllowSpeedUp": true,
      "DelayTimeAtDestination": 0
    },
    "Bulbasaur": {
      "MaxDistance": 500.0,
      "Priority": 2,
      "MaxWalkTimes": 300.0,
      "CatchPokemonWhileWalking": true,
      "SpinPokestopWhileWalking": true,
      "MaxSpeedUpSpeed": 6.0,
      "AllowSpeedUp": true,
      "DelayTimeAtDestination": 0
    },
    "Charmander": {
      "MaxDistance": 500.0,
      "Priority": 2,
      "MaxWalkTimes": 300.0,
      "CatchPokemonWhileWalking": true,
      "SpinPokestopWhileWalking": true,
      "MaxSpeedUpSpeed": 6.0,
      "AllowSpeedUp": true,
      "DelayTimeAtDestination": 0
    },
    "Squirtle": {
      "MaxDistance": 500.0,
      "Priority": 2,
      "MaxWalkTimes": 300.0,
      "CatchPokemonWhileWalking": true,
      "SpinPokestopWhileWalking": true,
      "MaxSpeedUpSpeed": 6.0,
      "AllowSpeedUp": true,
      "DelayTimeAtDestination": 0
    },
    "Omanyte": {
      "MaxDistance": 1500.0,
      "Priority": 3,
      "MaxWalkTimes": 300.0,
      "CatchPokemonWhileWalking": true,
      "SpinPokestopWhileWalking": true,
      "MaxSpeedUpSpeed": 6.0,
      "AllowSpeedUp": true,
      "DelayTimeAtDestination": 0
    },
    "Kabuto": {
      "MaxDistance": 1500.0,
      "Priority": 3,
      "MaxWalkTimes": 300.0,
      "CatchPokemonWhileWalking": true,
      "SpinPokestopWhileWalking": true,
      "MaxSpeedUpSpeed": 6.0,
      "AllowSpeedUp": true,
      "DelayTimeAtDestination": 0
    },
    "Marowak": {
      "MaxDistance": 1500.0,
      "Priority": 3,
      "MaxWalkTimes": 300.0,
      "CatchPokemonWhileWalking": true,
      "SpinPokestopWhileWalking": true,
      "MaxSpeedUpSpeed": 6.0,
      "AllowSpeedUp": true,
      "DelayTimeAtDestination": 0
    },
    "Vileplume": {
      "MaxDistance": 1500.0,
      "Priority": 3,
      "MaxWalkTimes": 300.0,
      "CatchPokemonWhileWalking": true,
      "SpinPokestopWhileWalking": true,
      "MaxSpeedUpSpeed": 6.0,
      "AllowSpeedUp": true,
      "DelayTimeAtDestination": 0
    }
```

### Steps to Reproduce
use a gpx file, use above settings and see for yourself

**Comment by samuraitruong** [comment]
well, GPX is having issue, I would like to spend few hours to investigate it after finish other work. can you send me a GPX file

**Comment by Silph-Road** [comment]
@samuraitruong 

Here is a sample GPX file:
Spawn near "-37,764294491535196" lon="145,35822987556458"
[GPXPath.zip](https://github.com/NoxxDev/NecroBot/files/456007/GPXPath.zip)

While you are looking at that, can you also investigate why the bot will jump/teleport to the next GPX point after it return back to starting gpx point, when it is human walk sniping.
Steps:

1. Bot scans sniper matches.
2. Bot chooses target.
3. Bot walks towards target (does not spin pokestops on the way)
4. Bot reaches target catches pokemon.
5. Bot starts to walk back to last gpx it visited
6. Bot reaches last gpx
7. Bot will try to catch pokemon at that location.
8. Bot will jump/teleport to next gpx point.

Please make sure that "AlwaysWalkback": false, otherwise you will jump/teleport the whole way back as referenced in [800](https://github.com/NoxxDev/NecroBot/issues/800)

Related issue: [Speed up not working, GPX](https://github.com/NoxxDev/NecroBot/issues/806)

**Comment by samuraitruong** [comment]
ok, will work on that tonight. that will very simple fix I guess

**Comment by Silph-Road** [comment]
@samuraitruong Awesome! Thanks :-)

**Comment by samuraitruong** [comment]
Checking code, I don't see anything wrong to make bot teleport .  @Silph-Road  have you test on the lastest code from master?

**Comment by Silph-Road** [comment]
@samuraitruong I am about to test the latest source right now. I will post back with results.

**Comment by samuraitruong** [comment]
The code of farmnearbypokestop is nasty. need to separate too. I  think i found the issue already.

**Comment by Silph-Road** [comment]
@samuraitruong That sounds great!

**Comment by samuraitruong** [comment]
the function to execute after reach destination  not being call if using GPX

**Comment by Silph-Road** [comment]
facepalm*

This fix will be great, cuz without it we run out of balls sooo fast. Especially if you have catch on the way on.

**Comment by samuraitruong** [comment]
I merged the code fix to master already. you can test it now.

**Comment by Silph-Road** [comment]
So I just tested it, and I think i see the pattern.

1. Bot reaches destination of snipe target pokemon
2. Bot catches pokemon
3. Bot starts to walk back towards last pokestop.

Now i have seen 3 possible things happen from the last step.

- Bot reaches last gpx point/pokestop it was at before sniping and then walks to next point.
OR
- Bot reaches the closes gpx point on its way back and then teleports to the next one.
OR
- Bot reaches last gpx point/pokestop it was at before sniping and then teleports to next point.

I think it is changing, based on the direction it is coming back from and the closest gpx location.

**Comment by Silph-Road** [comment]
Awesome ! @samuraitruong Disregard that last post. 
I will test with your latest source you just updated, and post results.

**Comment by Silph-Road** [comment]
@samuraitruong 

Its going to be a little bit before i cant post back, results from the new source.

But i did notice something..
When the bot is human walk sniping, the pokemon it catches on the way and the way back are not transferred immediately `("TransferDuplicatePokemonOnCapture": true,)`.

They all get transferred together (6-8) and looks very suspicious as most players transfer on capture, most realistic/human.

If the bot is just farming normally, they get transferred right after capture and follow the switch `("TransferDuplicatePokemonOnCapture": true,)` correctly.

**Comment by samuraitruong** [comment]
you want to create a pattern:)) 

event human, they not do transfer every pokemon they caught. I think that ok . during snipe session, transfer function will be disable.

**Comment by Silph-Road** [comment]
yeah its not a big deal, makes it a bit random! :-)

**Comment by Silph-Road** [comment]
Another thing i just noticed, I dont know if its GPX or Human walk related. 

- Pokestop visted do not show as cooled down after 5 mins period, if `"CatchPokemon": true,`
- If `"CatchPokemon": false,` they show cooled down in visualizer.

**Comment by Capusjon** [comment]
@samuraitruong 
@Silph-Road 
Thanks for picking this up guys!

I'll test with the latest source as well, if I experience anything wrong I'll report back.

What should the bot be doing if it has reached the snipetarget and is done catching? I assume it should first check next snipe target if no target is available it should pick up the nearest gpx point and continue the route from there. At least that is what I would do if I was walking IRL.

**Comment by Capusjon** [comment]
Mabey an other issue, but partially why it won't make it to all snipetargets:

If you look at the settings beneath I would expect the bot to calculate snipetargets at 5.2 speedup since it's the lowest determining variable. It uses 6 though. I can work arround by giving all pokemon a 5.2 speedup but I don't think this is the way it should work.

  "HumanWalkSnipeConfig": {
"MaxSpeedUpSpeed": 5.2,

in pokemonspecific it states 6km/h
Kabuto": {
      "MaxDistance": 1500.0,
      "Priority": 3,
      "MaxWalkTimes": 300.0,
      "CatchPokemonWhileWalking": true,
      "SpinPokestopWhileWalking": true,
      "MaxSpeedUpSpeed": 6.0,
      "AllowSpeedUp": true,
      "DelayTimeAtDestination": 0

**Comment by Capusjon** [comment]
I can confirm it picks up the nearest gpx point sometimes.

**Comment by Silph-Road** [comment]
@Capusjon 
> What should the bot be doing if it has reached the snipetarget and is done catching? I assume it should first check next snipe target if no target is available it should pick up the nearest gpx point and continue the route from there. At least that is what I would do if I was walking IRL.

If you have `"TryCatchEmAll": false,` it will walk back to origin.
If you have `"TryCatchEmAll": true,` it will move to next target (I think this target was found before starting the first snipe).

@samuraitruong Is this correct?

**Comment by Silph-Road** [comment]
@Capusjon I think @samuraitruong is still working on the the pokestop loot on the way to target and way back.

**Comment by samuraitruong** [comment]
That is another story. Will fix after release gym feature

Get Outlook for Android<https://aka.ms/ghei36>



On Wed, Sep 7, 2016 at 7:49 AM +1000, "Silph-Road" <notifications@github.com<mailto:notifications@github.com>> wrote:


@Capusjon<https://github.com/Capusjon> I think @samuraitruong<https://github.com/samuraitruong> is still working on the the pokestop loot on the way to target and way back.
